src/store.cc

```diff
@@ -1537,7 +1537,12 @@ StoreEntry::updateOnNotModified(const StoreEntry &e304)
     if (!timestampsSet() && !updatedReply)
         return false;
 
-    // XXX: Update old->mem_obj->vary_headers?
+    assert(e304.mem_obj->request);
+    const auto updatedVaryMark = updatedReply->makeVaryMark(*(e304.mem_obj->request));
```

Comment:
`updatedReply` may be nil here (if we updated nothing in the reply). Refactor to skip Vary update in that case?

Even after that refactoring, AFAICT, you want to use the freshestReply here, not updatedReply and not baseReply.